samples/builtin-kernel-example/builtin-kernel-example.cpp

```diff
@@ -0,0 +1,119 @@
+/***************************************************************************
+ *
+ *  Copyright (C) 2016 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's ComputeCpp SDK
+ *
+ *  builtin_kernel_example.cpp
+ *
+ *  Description:
+ *    Example of using an OpenCL builtin kernel with SYCL via the codeplay
+ *    extension
+ *
+ **************************************************************************/
+
+#include <CL/sycl.hpp>
+
+#include <array>
+#include <iostream>
+
+int main() {
+
+  queue testQueue(TEST_SELECTOR, TEST_ASYNC_HANDLER);
+  context testContext = testQueue.get_context();
+
+  device dev = testContext.get_devices()[0];
+
+  /*
+   * Using the device properties, we can query which built in kernels
+   * are supported
+   */
+  auto builtinKernels = dev.get_info<info::device::built_in_kernels>();
+
+  if (builtinKernels.size() == 0) {
+     std::cout << "[EXIT] No built-in kernels available for testing " 
+               << std::endl;
+      return 0;
+  }
+
+  /*
+   * The only builtin kernel supported by this example is the
+   * ComputeAorta copy buffer, defined as:
+   *    copy_buffer(__global * in, __global * out)
+   * And copies input into output.
+   */
+  const std::string kAortaTestKernelName{"copy_buffer"};
```

Comment:
Not super keen on the `k` for constant naming convention, but it's no big deal. `copy_buffer_kernel` might be more informative (though perhaps you want to preserve the ComputeAorta name?).

Reply:
I have to preserve that kernel name or it cannot be tested on Aorta.

Reply:
I meant the name of the variable sorry, not the string it is containing - again, not a necessary change!